pkg/controller/job/job_controller.go

```diff
@@ -651,7 +655,9 @@ func (jm *Controller) syncJob(key string) (forget bool, rErr error) {
 		if isIndexedJob(&job) {
 			job.Status.CompletedIndexes = succeededIndexes
 		}
-
+		if err := jm.ensureStartTime(&job, setStartTime); err != nil {
```

Comment:
I don't think this solves your problem, what you need to do is to change the condition above which ensures that `updateHandler` is called at all.

Reply:
@soltysh Can you give a sample code to illustrate what should change for the condition? The root cause for the problem is the job.Status.starttime being set twice, but I don't know how to control calling updateHandler by conditions for this problem, thanks

Reply:
I'm thinking of something like:
```golang
if job.Status.Active != active || job.Status.Succeeded != succeeded || job.Status.Failed != failed || jobConditionsChanged  || setStartTime {
```
and then make sure that flag and time are properly set at start and resume.

Reply:
@soltysh If using this condition to control, it maybe lose some job status update, because only starttime mistake, others job status corrrect. so I still think rechecking the starttime in updatehandler is a good catch

Reply:
Right, but the problem is unless something else gets changed it won't enter that condition at all.

Reply:
Yea, so if for example  job.Status.Active != active  is right, it will ignore the following conditions, the starttime may still be updated again, so I add ensureStartTime in the IF Statement{} to prevent starttime being re-set as a result of race condition

Reply:
@soltysh This problem is a race condition as two job-workers (go-routines) update job status all contains similar starttime just in time. so inside a worker's sync, we could not know whether another job has just updated the job status with a short early starttime, we only can check this as nearest as writing to storage. Only get the fresh job from storage at the latest time, I could see whether job starttime need reset

Reply:
@soltysh I know what you said, I will push again later, thanks